agent/api/statechange.go

```diff
@@ -171,36 +171,30 @@ func newUncheckedContainerStateChangeEvent(task *apitask.Task, cont *apicontaine
 		ImageDigest:   cont.GetImageDigest(),
 		Reason:        reason,
 		Container:     cont,
-		Type:          eventType,
 	}
 	return event, nil
 }
 
-// NewManagedAgentChangeEvent creates a new special container state change event to convey managed agent state changes
+// NewManagedAgentChangeEvent creates a new managedAgent change event to convey managed agent state changes
 // returns error if the state change doesn't need to be sent to the ECS backend.
-func NewManagedAgentChangeEvent(task *apitask.Task, cont *apicontainer.Container) (ContainerStateChange, error) {
-	var changedManagedAgents []ManagedAgentStateChange
-	for _, ma := range cont.GetManagedAgents() {
-		if ma.Status.ShouldReportToBackend() {
-			changedManagedAgents = append(changedManagedAgents, ManagedAgentStateChange{
-				Name:   ma.Name,
-				Status: ma.Status,
-				Reason: ma.Reason,
-			})
-		}
+func NewManagedAgentChangeEvent(task *apitask.Task, cont *apicontainer.Container, managedAgentName string, reason string) (ManagedAgentStateChange, error) {
```

Comment:
missing unit tests

Reply:
agent/api/statechange_test.go has updated:
`func TestNewManagedAgentChangeEvent(t *testing.T) {`

Reply:
added case for non-existent managed agent.